batavia/VirtualMachine.js

```diff
@@ -1357,12 +1357,16 @@ batavia.VirtualMachine.prototype.byte_DELETE_ATTR = function(name) {
 
 batavia.VirtualMachine.prototype.byte_STORE_SUBSCR = function() {
     var items = this.popn(3);
-    items[1][items[2]] = items[0];
+    if (items[1].__setitem__) {
+        items[1].__setitem__(items[2], items[0]);
+    } else {
+        items[1][items[2]] = items[0];
+    }
 };
 
 batavia.VirtualMachine.prototype.byte_DELETE_SUBSCR = function() {
     var items = this.popn(2);
-    delete items[1][items[0]];
+    items[1].__delitem__(items[0]);
```

Comment:
Shouldn't this have the same fallback conditions as STORE_SUBSCR?

Or, alternatively, STORE_SUBSCR should be simplified to just use the Pythonic API.